File: src/test/java/aoc2015/day01/Day01Test.java
package aoc2015.day01;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Day01Test {

    @Test
    void part1Test() {
        var day = new Day01();

        assertEquals(0, day.part1("(())"));
        assertEquals(3, day.part1("(()(()("));
        assertEquals(3, day.part1("))((((("));
        assertEquals(-1, day.part1("))("));
        assertEquals(-3, day.part1(")())())"));
    }

    @Test
    void part2Test() {
        var day = new Day01();

        assertEquals(1, day.part2(")"));
        assertEquals(5, day.part2("()())"));
    }
}

File: src/test/java/aoc2020/day01/Day01Test.kt
package aoc2020.day01

import aoc.loadExample
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class Day01Test {

    @Test
    fun part1Example() {
        val input = loadExample(2020, "day01.txt")
        val subject = Day01()

        val result = subject.part1(input)

        assertThat(result)
            .isEqualTo(514579)
    }

    @Test
    fun part2Example() {
        val input = loadExample(2020, "day01.txt")
        val subject = Day01()

        val result = subject.part2(input)

        assertThat(result)
            .isEqualTo(241861950)
    }
}